src/App.tsx

```diff
@@ -14,11 +16,120 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[],
+};
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  title = () => {
+    const { selectedGoods } = this.state;
+    let message;
+
+    if (selectedGoods.length === 0) {
+      message = 'No goods selected';
+    }
+
+    if (selectedGoods.length === 1) {
+      message = `${selectedGoods[0]} is selected`;
+    }
+
+    if (selectedGoods.length >= 2) {
+      const firstPart = selectedGoods.slice(0, selectedGoods.length - 1);
+      const secondPart = selectedGoods[selectedGoods.length - 1];
+
+      message = `${firstPart.join(', ')} and ${secondPart} are selected`;
+    }
+
+    return message;
+  };
+
+  selectedItem = (item: string) => {
+    this.setState((prevState) => ({
+      selectedGoods: [...prevState.selectedGoods, item],
+    }));
+  };
+
+  removedItem = (item: string) => {
+    this.setState((prevState) => ({
+      selectedGoods: prevState.selectedGoods.filter(good => good !== item),
+    }));
+  };
+
+  getRelevantFunction = (item: string) => {
+    if (this.state.selectedGoods.includes(item)) {
+      this.removedItem(item);
+    } else {
+      this.selectedItem(item);
+    }
+  };
```

Comment:
I think you don't need this function

Reply:
Do you suggest using the "if" statement right in onClick attribute or smth elese?